src/profile/helpers/editData.ts

```diff
@@ -0,0 +1,202 @@
+import _ from 'lodash';
+
+import {
+  ProfileRoot,
+  ProfileData,
+  AddressNode,
+  EmailNode,
+  PhoneNode,
+  PrimaryAddress,
+  PrimaryEmail,
+  PrimaryPhone,
+  Language,
+} from '../../graphql/typings';
+import { formConstants } from '../constants/formConstants';
+import getAddressesFromNode from '../helpers/getAddressesFromNode';
+import getEmailsFromNode from '../helpers/getEmailsFromNode';
+import getPhonesFromNode from '../helpers/getPhonesFromNode';
+
+export type BasicDataValue = Pick<
+  ProfileData,
+  'firstName' | 'nickname' | 'lastName'
+>;
+export type AdditionalInformationValue = {
+  language: Language;
+};
+export type AddressValue = Pick<
+  AddressNode,
+  'address' | 'city' | 'postalCode' | 'countryCode'
+>;
+
+export type EmailValue = Pick<EmailNode, 'email'>;
+export type PhoneValue = Pick<PhoneNode, 'phone'>;
+
+type BasicDataSource = BasicDataValue & { id: ProfileData['id'] };
+type AdditionalInformationSource = Pick<ProfileData, 'id' | 'language'>;
+export type EditDataProfileSource =
+  | BasicDataSource
+  | AdditionalInformationSource
+  | AddressNode
+  | PhoneNode
+  | EmailNode;
+
+export type MultiItemProfileNode = AddressNode | PhoneNode | EmailNode;
+
+export const basicDataType = 'basic-data';
+export const additionalInformationType = 'additional-information';
+
+export type EditDataType =
+  | 'phones'
+  | 'emails'
+  | 'addresses'
+  | typeof basicDataType
+  | typeof additionalInformationType;
+
+export type EditDataValue =
+  | BasicDataValue
+  | AdditionalInformationValue
+  | AddressValue
+  | PhoneValue
+  | EmailValue;
+
+export type FormValues = {
+  firstName: string;
+  nickname: string;
+  lastName: string;
+  primaryEmail: PrimaryEmail;
+  primaryAddress: PrimaryAddress;
+  primaryPhone: PrimaryPhone;
```

Comment:
Answered better in the other primary-related comments. Perhaps  these are obsolete at the moment? Why were they ever added/needed? 😄

Reply:
Not sure, but perhaps for such clients that only need e.g. one email address so that they can send some email to the customer. So a short hand to avoid searching through the whole list.